Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/ScriptCanvasAutoGenRegistry.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <algorithm>
+
+#include "ScriptCanvasAutoGenRegistry.h"
+
+namespace ScriptCanvas
+{
+    AutoGenRegistry::~AutoGenRegistry()
+    {
+        m_functions.clear();
+    }
+
+    AutoGenRegistry* AutoGenRegistry::GetInstance()
+    {
+        // Use static object so each module will keep its own registry collection
+        static AutoGenRegistry s_autogenRegistry;
+
+        return &s_autogenRegistry;
+    }
+
+    void AutoGenRegistry::ReflectFunctions(AZ::ReflectContext* context)
```

Comment:
I would just call it `Reflect` for consistency

Reply:
The initial purpose is to distinguish different reflect logic, like later we will add nodeable node reflection into this registry as well.
But yes, will create a Reflect function as public, and ReflectFunctions as private one.